float64[] pos
float64[] rot
---
